src/polyfem/assembler/RhsAssembler.cpp

```diff
@@ -729,7 +729,7 @@ namespace polyfem
 			}
 		}
 
-		double RhsAssembler::compute_energy(const Eigen::MatrixXd &displacement, const std::vector<LocalBoundary> &local_neumann_boundary, const Density &density, const int resolution, const double t) const
+		double RhsAssembler::compute_energy(const Eigen::MatrixXd &displacement, const Eigen::MatrixXd &displacement_prev, const std::vector<LocalBoundary> &local_neumann_boundary, const Density &density, const int resolution, const double t) const
```

Comment:
do we need displacement?

so we use the prev for the normals, but displacement for the rest?

Reply:
Yes, so the normal computation for the pressure is lagged but the energy needs the current displacement as the gradient is constant.